src/Form/DataTransformer/ArrayToModelTransformer.php

```diff
@@ -30,35 +32,57 @@ class ArrayToModelTransformer implements DataTransformerInterface
 
     /**
      * @var string
+     *
+     * @phpstan-var class-string<T>
      */
     protected $className;
 
     /**
      * @param string $className
+     *
+     * @phpstan-param class-string<T> $className
      */
     public function __construct(ModelManagerInterface $modelManager, $className)
     {
         $this->modelManager = $modelManager;
         $this->className = $className;
     }
 
-    public function reverseTransform($array)
+    /**
+     * @param object|object[]|null $value
+     *
+     * @return object
+     *
+     * @phpstan-param T|T[]|null $value
+     *
+     * @phpstan-return T
+     */
+    public function reverseTransform($value)
     {
         // when the object is created the form return an array
         // one the object is persisted, the edit $array is the user instance
-        if ($array instanceof $this->className) {
-            return $array;
+        if ($value instanceof $this->className) {
+            return $value;
         }
 
         $instance = new $this->className();
 
-        if (!\is_array($array)) {
+        if (!\is_array($value)) {
             return $instance;
         }
 
-        return $this->modelManager->modelReverseTransform($this->className, $array);
+        return $this->modelManager->modelReverseTransform($this->className, $value);
     }
 
+    /**
+     * @param mixed|null $value
```

Comment:
If you use `@phpstan-param T|null $value`, you should use `@param object|null $value`

Same for `@return`.